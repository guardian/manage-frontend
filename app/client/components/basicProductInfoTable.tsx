import { parseDate } from '../../shared/dates';
import { ProductDetail } from '../../shared/productResponse';
import { GroupedProductType } from '../../shared/productTypes';
import { ProductDescriptionListTable } from './productDescriptionListTable';

interface BasicProductInfoTableProps {
	groupedProductType: GroupedProductType;
	productDetail: ProductDetail;
}

export const BasicProductInfoTable = (props: BasicProductInfoTableProps) => {
	return (
		<ProductDescriptionListTable
			content={[
				...(props.groupedProductType.shouldRevealSubscriptionId
					? [
							{
								title: 'Subscription ID',
								value: props.productDetail.subscription
									.subscriptionId,
							},
					  ]
					: []),
				...(props.groupedProductType.tierLabel
					? [
							{
								title: props.groupedProductType.tierLabel,
								value: props.productDetail.tier,
							},
					  ]
					: []),
				...(props.groupedProductType.shouldShowJoinDateNotStartDate
					? [
							{
								title: 'Join date',
								value: parseDate(
									props.productDetail.joinDate,
								).dateStr(),
							},
					  ]
					: [
							{
								title: 'Start date',
								value: props.productDetail.subscription.start
									? parseDate(
											props.productDetail.subscription
												.start,
									  ).dateStr()
									: '-',
							},
					  ]),
			]}
		/>
	);
};
