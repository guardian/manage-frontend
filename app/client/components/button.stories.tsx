import React from "react";
import { ComponentStory, ComponentMeta } from "@storybook/react";

import { Button, ButtonProps } from "./buttons";

export default {
  title: "Components/Button",
  component: Button,
  args: {
    text: "Button",
    height: "",
    fontWeight: undefined,
    left: false,
    right: false,
    disabled: false,
    colour: undefined,
    textColour: undefined,
    primary: undefined,
    hide: false,
    forceCircle: undefined,
    hoverColour: undefined,
    leftTick: undefined,
    alert: false,
    type: "button",
  },
  argTypes: {
    fontWeight: {
      options: ["bold"],
    },
    primary: {
      control: { type: "boolean" },
    },
    hollow: {
      control: { type: "boolean" },
    },
    forceCircle: {
      control: { type: "boolean" },
    },
    leftTick: {
      control: { type: "boolean" },
    },
    colour: {
      control: { type: "color" },
    },
    textColour: {
      control: { type: "color" },
    },
    hoverColour: {
      control: { type: "color" },
    },
  },
} as ComponentMeta<typeof Button>;

const Template: ComponentStory<typeof Button> = (args: ButtonProps) => (
  <Button {...args} />
);

export const Default = Template.bind({});

export const Primary = Template.bind({});
Primary.args = {
  primary: true,
};
