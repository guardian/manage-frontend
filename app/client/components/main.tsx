import React from 'react';
import palette from '../colours';
import { serif } from '../styles/fonts';
import { Footer } from './footer/footer';
import Header from './header';
import { SignInStatus } from '../services/signInStatus';
import HelpCentreHeader from './helpCentre/helpCentreHeader';

export interface MainProps {
	signInStatus?: SignInStatus;
	requiresSignIn?: boolean;
	helpCentrePage?: boolean;
	children: JSX.Element | JSX.Element[];
}

export const Main = ({
	signInStatus = 'init',
	requiresSignIn,
	helpCentrePage,
	children,
}: MainProps) => (
	<div
		css={{
			display: 'flex',
			flexDirection: 'column',
			height: '100vh',
			alignItems: 'stretch',
			width: '100%',
			color: palette.neutral['2'],
		}}
	>
		{helpCentrePage ? (
			<HelpCentreHeader
				signInStatus={signInStatus}
				requiresSignIn={requiresSignIn}
			/>
		) : (
			<Header
				signInStatus={signInStatus}
				requiresSignIn={requiresSignIn}
			/>
		)}
		<div
			css={{
				flexGrow: 1,
				flexShrink: 0,
				display: 'flex',
				flexDirection: 'column',
			}}
		>
			<main
				css={{
					fontFamily: serif,
					flexGrow: 1,
					flexShrink: 0,
				}}
			>
				{children}
			</main>
		</div>
		<Footer />
	</div>
);
