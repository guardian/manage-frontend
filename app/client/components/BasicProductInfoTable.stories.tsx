import React from 'react';
import { ComponentStory, ComponentMeta } from '@storybook/react';

import { BasicProductInfoTable } from './basicProductInfoTable';
import { newspaperVoucherPaypal } from '../fixtures/productDetail';
import { GROUPED_PRODUCT_TYPES } from '../../shared/productTypes';

export default {
	title: 'Components/BasicProductInfoTable',
	component: BasicProductInfoTable,
	parameters: {
		controls: { hideNoControlsWarning: true },
	},
} as ComponentMeta<typeof BasicProductInfoTable>;

export const Default: ComponentStory<typeof BasicProductInfoTable> = () => (
	<BasicProductInfoTable
		groupedProductType={GROUPED_PRODUCT_TYPES.subscriptions}
		productDetail={newspaperVoucherPaypal}
	/>
);
