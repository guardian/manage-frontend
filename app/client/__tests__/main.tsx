import React from "react";
import serializer from "jest-emotion";
import renderer from "react-test-renderer";
import { Main } from "../components/main";

expect.addSnapshotSerializer(serializer);

describe("Main", () => {
  beforeEach(() => {
    jest.useFakeTimers();
    jest.setSystemTime(new Date("2022-01-01"));
  });

  afterEach(() => {
    jest.useRealTimers();
  });

  it("renders something", () => {
    const rendered = renderer.create(
      <Main>
        <p>hi</p>
      </Main>
    );

    expect(rendered.toJSON()).toMatchSnapshot();
  });
});
