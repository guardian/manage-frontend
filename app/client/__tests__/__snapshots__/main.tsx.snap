// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main renders something 1`] = `
<div
  css={
    Object {
      "alignItems": "stretch",
      "color": "#333333",
      "display": "flex",
      "flexDirection": "column",
      "height": "100vh",
      "width": "100%",
    }
  }
>
  <header
    css={
      Object {
        "@media (min-width: 980px)": Object {
          "minHeight": "82px",
        },
        "backgroundColor": "#052962",
        "boxShadow": "0 2px 1px -1px #506991",
        "minHeight": "50px",
        "overflow": "visible",
        "position": "relative",
        "zIndex": 1070,
      }
    }
  >
    <div
      css={
        Object {
          "@media (min-width: 1300px)": Object {
            "gridTemplateColumns": "repeat(16, minmax(0, 1fr))",
            "msGridColumns": "(minmax(0, 1fr))[16]",
          },
          "@media (min-width: 740px)": Object {
            "gridTemplateColumns": "repeat(12, minmax(0, 1fr))",
            "msGridColumns": "(minmax(0, 1fr))[12]",
            "paddingLeft": "20px",
            "paddingRight": "20px",
          },
          "@supports (display: grid)": Object {
            "display": "grid",
          },
          "alignItems": "center",
          "columnGap": "20px",
          "display": "-ms-grid",
          "gridAutoColumns": "max-content",
          "gridTemplateColumns": "repeat(4, minmax(0, 1fr))",
          "height": "100%",
          "margin": "auto",
          "maxWidth": "calc(1300px + 2.5rem)",
          "msGridColumns": "(minmax(0, 1fr))[4]",
          "paddingLeft": "12px",
          "paddingRight": "12px",
        }
      }
    >
      <div
        css={
          Object {
            "@media (min-width: 1300px)": Object {
              "gridColumnEnd": "span 1",
              "gridColumnStart": "-2",
              "msGridColumn": "16",
              "msGridColumnSpan": "1",
              "msGridRow": "1",
            },
            "@media (min-width: 740px)": Object {
              "gridColumnEnd": "span 1",
              "gridColumnStart": "-2",
              "margin": "auto",
              "maxHeight": "51px",
              "msGridColumn": "12",
              "msGridColumnSpan": "1",
              "msGridRow": "1",
            },
            "display": "inline-block",
            "gridColumnEnd": "span 1",
            "gridColumnStart": "-2",
            "msGridColumn": "4",
            "msGridColumnSpan": "1",
            "msGridRow": "1",
          }
        }
      >
        <a
          css={
            Object {
              "@media (min-width: 980px)": Object {
                "height": "51px",
                "width": "51px",
              },
              "display": "block",
              "height": "39px",
              "margin": "auto 0 auto auto",
              "textAlign": "right",
            }
          }
          href="https://www.theguardian.com"
          title="The Guardian - Back to home"
        >
          <svg
            css={
              Object {
                "@media (min-width: 980px)": Object {
                  "height": "51px",
                  "width": "51px",
                },
                "height": "39px",
                "width": "39px",
              }
            }
            viewBox="0 0 56 56"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M28 0a28 28 0 1 0 28 28A28 28 0 0 0 28 0"
              fill="#FFFFFF"
            />
            <path
              d="M33 6.92c3.63.58 8.24 3.06 9.89 4.83v8h-1L33 7.82zm-3.34.5h-.09c-6.35 0-9.8 8.8-9.8 20.66 0 11.87 3.45 20.66 9.8 20.66h.09v.91c-9.56.65-22.42-6.63-22.09-21.58C7.23 13.14 20.09 5.86 29.66 6.51zm16.16 22.53l-3 1.3v13.44A24.26 24.26 0 0 1 33 49.52V31l-3.3-1.09V29h16.12z"
              fill="#052962"
            />
          </svg>
        </a>
      </div>
    </div>
  </header>
  <div
    css={
      Object {
        "display": "flex",
        "flexDirection": "column",
        "flexGrow": 1,
        "flexShrink": 0,
      }
    }
  >
    <main
      css={
        Object {
          "flexGrow": 1,
          "flexShrink": 0,
          "fontFamily": "\\"GuardianTextEgyptian\\", Georgia, serif",
        }
      }
    >
      <p>
        hi
      </p>
    </main>
  </div>
  <footer>
    <div>
      <div
        css={
          Object {
            "map": undefined,
            "name": "vck559",
            "next": undefined,
            "styles": "
  background-color: #052962;
  color: #ffffff;
",
            "toString": [Function],
          }
        }
      >
        <div
          css={
            Object {
              "map": undefined,
              "name": "1fkn892",
              "next": undefined,
              "styles": "
  max-width: 1300px;
  margin: auto;
",
              "toString": [Function],
            }
          }
        >
          <div
            css={
              Object {
                "map": undefined,
                "name": "1ku4d2x",
                "next": undefined,
                "styles": "
  padding: 10px;
  border: 1px solid rgba(255, 255, 255, 0.3);
  border-top: 0;

  @media (min-width: 980px) {
    padding: 0 20px;
  }

  @media (min-width: 1140px) {
    display: flex;
  }
",
                "toString": [Function],
              }
            }
          >
            <div
              css={
                Object {
                  "map": undefined,
                  "name": "1jqhtyp",
                  "next": undefined,
                  "styles": "
  padding: 0;
  border: 0;
  width: 100%;
  margin: 0;
  @media (min-width: 1140px) {
    width: 340px;
  }
  @media (min-width: 1300px) {
    width: 460px;
  }
",
                  "toString": [Function],
                }
              }
            >
              <iframe
                css={
                  Object {
                    "map": undefined,
                    "name": "a2zj5o",
                    "next": undefined,
                    "styles": "
  min-height: 150px;
",
                    "toString": [Function],
                  }
                }
                data-form-success-desc="We will send you our picks of the most important headlines tomorrow morning."
                data-node-uid="2"
                frameBorder="0"
                height="86px"
                onLoad={[Function]}
                scrolling="no"
                seamless={false}
                src="https://www.theguardian.com/email/form/footer/today-uk"
                title="Guardian Email Sign-up Form"
              />
            </div>
            <div
              css={
                Object {
                  "map": undefined,
                  "name": "1x6vqcu",
                  "next": undefined,
                  "styles": "
  font-feature-settings: kern;
  font-size: 16px;
  line-height: 16px;
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  padding-bottom: 18px;
  border-top: 1px solid rgba(255, 255, 255, 0.3);
  @media (min-width: 1300px) {
    border-top: 0;
  }
",
                  "toString": [Function],
                }
              }
            >
              <ul
                css={
                  Object {
                    "map": undefined,
                    "name": "1hkg2v9",
                    "next": undefined,
                    "styles": "
  line-height: 19.2px;
  width: calc(50% - 1.25rem - 1px);
  list-style: none;
  position: relative;
  padding: 0 10px;
  margin: 0;

  &:nth-of-type(even) {
    border-left: 1px solid rgba(255, 255, 255, 0.3);
  }

  @media (min-width: 740px) {
    &:not(:first-of-type) {
      border-left: 1px solid rgba(255, 255, 255, 0.3);
    }

    width: 161px;
    flex: 1 0 0;
  }

  @media (min-width: 980px) {
    border-left: 1px solid rgba(255, 255, 255, 0.3);
  }
",
                    "toString": [Function],
                  }
                }
              >
                <li
                  css={
                    Object {
                      "map": undefined,
                      "name": "66iazq",
                      "next": undefined,
                      "styles": "
  list-style: none;
",
                      "toString": [Function],
                    }
                  }
                >
                  <a
                    css={
                      Object {
                        "map": undefined,
                        "name": "1x52xav",
                        "next": undefined,
                        "styles": "
  display: inline-block;
  padding: 6px 0;
  color: #ffffff;
  :hover {
    color: #ffe500;
    cursor: pointer;
  }
",
                        "toString": [Function],
                      }
                    }
                    href="https://thegulocal.com/about"
                  >
                    About us
                  </a>
                </li>
                <li
                  css={
                    Object {
                      "map": undefined,
                      "name": "66iazq",
                      "next": undefined,
                      "styles": "
  list-style: none;
",
                      "toString": [Function],
                    }
                  }
                >
                  <a
                    css={
                      Object {
                        "map": undefined,
                        "name": "1x52xav",
                        "next": undefined,
                        "styles": "
  display: inline-block;
  padding: 6px 0;
  color: #ffffff;
  :hover {
    color: #ffe500;
    cursor: pointer;
  }
",
                        "toString": [Function],
                      }
                    }
                    href="https://thegulocal.com/help/contact-us"
                  >
                    Contact us
                  </a>
                </li>
                <li
                  css={
                    Object {
                      "map": undefined,
                      "name": "66iazq",
                      "next": undefined,
                      "styles": "
  list-style: none;
",
                      "toString": [Function],
                    }
                  }
                >
                  <a
                    css={
                      Object {
                        "map": undefined,
                        "name": "1x52xav",
                        "next": undefined,
                        "styles": "
  display: inline-block;
  padding: 6px 0;
  color: #ffffff;
  :hover {
    color: #ffe500;
    cursor: pointer;
  }
",
                        "toString": [Function],
                      }
                    }
                    href="https://thegulocal.com/info/complaints-and-corrections"
                  >
                    Complaints & corrections
                  </a>
                </li>
                <li
                  css={
                    Object {
                      "map": undefined,
                      "name": "66iazq",
                      "next": undefined,
                      "styles": "
  list-style: none;
",
                      "toString": [Function],
                    }
                  }
                >
                  <a
                    css={
                      Object {
                        "map": undefined,
                        "name": "1x52xav",
                        "next": undefined,
                        "styles": "
  display: inline-block;
  padding: 6px 0;
  color: #ffffff;
  :hover {
    color: #ffe500;
    cursor: pointer;
  }
",
                        "toString": [Function],
                      }
                    }
                    href="https://thegulocal.com/securedrop"
                  >
                    Secure Drop
                  </a>
                </li>
                <li
                  css={
                    Object {
                      "map": undefined,
                      "name": "66iazq",
                      "next": undefined,
                      "styles": "
  list-style: none;
",
                      "toString": [Function],
                    }
                  }
                >
                  <a
                    css={
                      Object {
                        "map": undefined,
                        "name": "1x52xav",
                        "next": undefined,
                        "styles": "
  display: inline-block;
  padding: 6px 0;
  color: #ffffff;
  :hover {
    color: #ffe500;
    cursor: pointer;
  }
",
                        "toString": [Function],
                      }
                    }
                    href="https://workforus.thegulocal.com"
                  >
                    Work for us
                  </a>
                </li>
                <li
                  css={
                    Object {
                      "map": undefined,
                      "name": "66iazq",
                      "next": undefined,
                      "styles": "
  list-style: none;
",
                      "toString": [Function],
                    }
                  }
                >
                  <a
                    css={
                      Object {
                        "map": undefined,
                        "name": "1x52xav",
                        "next": undefined,
                        "styles": "
  display: inline-block;
  padding: 6px 0;
  color: #ffffff;
  :hover {
    color: #ffe500;
    cursor: pointer;
  }
",
                        "toString": [Function],
                      }
                    }
                    href="#"
                    onClick={[Function]}
                  >
                    Privacy settings
                  </a>
                </li>
                <li
                  css={
                    Object {
                      "map": undefined,
                      "name": "66iazq",
                      "next": undefined,
                      "styles": "
  list-style: none;
",
                      "toString": [Function],
                    }
                  }
                >
                  <a
                    css={
                      Object {
                        "map": undefined,
                        "name": "1x52xav",
                        "next": undefined,
                        "styles": "
  display: inline-block;
  padding: 6px 0;
  color: #ffffff;
  :hover {
    color: #ffe500;
    cursor: pointer;
  }
",
                        "toString": [Function],
                      }
                    }
                    href="https://thegulocal.com/info/privacy"
                  >
                    Privacy policy
                  </a>
                </li>
                <li
                  css={
                    Object {
                      "map": undefined,
                      "name": "66iazq",
                      "next": undefined,
                      "styles": "
  list-style: none;
",
                      "toString": [Function],
                    }
                  }
                >
                  <a
                    css={
                      Object {
                        "map": undefined,
                        "name": "1x52xav",
                        "next": undefined,
                        "styles": "
  display: inline-block;
  padding: 6px 0;
  color: #ffffff;
  :hover {
    color: #ffe500;
    cursor: pointer;
  }
",
                        "toString": [Function],
                      }
                    }
                    href="https://thegulocal.com/info/cookies"
                  >
                    Cookie policy
                  </a>
                </li>
                <li
                  css={
                    Object {
                      "map": undefined,
                      "name": "66iazq",
                      "next": undefined,
                      "styles": "
  list-style: none;
",
                      "toString": [Function],
                    }
                  }
                >
                  <a
                    css={
                      Object {
                        "map": undefined,
                        "name": "1x52xav",
                        "next": undefined,
                        "styles": "
  display: inline-block;
  padding: 6px 0;
  color: #ffffff;
  :hover {
    color: #ffe500;
    cursor: pointer;
  }
",
                        "toString": [Function],
                      }
                    }
                    href="https://www.thegulocal.com/help/terms-of-service"
                  >
                    Terms & conditions
                  </a>
                </li>
                <li
                  css={
                    Object {
                      "map": undefined,
                      "name": "66iazq",
                      "next": undefined,
                      "styles": "
  list-style: none;
",
                      "toString": [Function],
                    }
                  }
                >
                  <a
                    css={
                      Object {
                        "map": undefined,
                        "name": "1x52xav",
                        "next": undefined,
                        "styles": "
  display: inline-block;
  padding: 6px 0;
  color: #ffffff;
  :hover {
    color: #ffe500;
    cursor: pointer;
  }
",
                        "toString": [Function],
                      }
                    }
                    href="https://www.thegulocal.com/help"
                  >
                    Help
                  </a>
                </li>
              </ul>
              <ul
                css={
                  Object {
                    "map": undefined,
                    "name": "1hkg2v9",
                    "next": undefined,
                    "styles": "
  line-height: 19.2px;
  width: calc(50% - 1.25rem - 1px);
  list-style: none;
  position: relative;
  padding: 0 10px;
  margin: 0;

  &:nth-of-type(even) {
    border-left: 1px solid rgba(255, 255, 255, 0.3);
  }

  @media (min-width: 740px) {
    &:not(:first-of-type) {
      border-left: 1px solid rgba(255, 255, 255, 0.3);
    }

    width: 161px;
    flex: 1 0 0;
  }

  @media (min-width: 980px) {
    border-left: 1px solid rgba(255, 255, 255, 0.3);
  }
",
                    "toString": [Function],
                  }
                }
              >
                <li
                  css={
                    Object {
                      "map": undefined,
                      "name": "66iazq",
                      "next": undefined,
                      "styles": "
  list-style: none;
",
                      "toString": [Function],
                    }
                  }
                >
                  <a
                    css={
                      Object {
                        "map": undefined,
                        "name": "1x52xav",
                        "next": undefined,
                        "styles": "
  display: inline-block;
  padding: 6px 0;
  color: #ffffff;
  :hover {
    color: #ffe500;
    cursor: pointer;
  }
",
                        "toString": [Function],
                      }
                    }
                    href="https://thegulocal.com/index/subjects/a"
                  >
                    All topics
                  </a>
                </li>
                <li
                  css={
                    Object {
                      "map": undefined,
                      "name": "66iazq",
                      "next": undefined,
                      "styles": "
  list-style: none;
",
                      "toString": [Function],
                    }
                  }
                >
                  <a
                    css={
                      Object {
                        "map": undefined,
                        "name": "1x52xav",
                        "next": undefined,
                        "styles": "
  display: inline-block;
  padding: 6px 0;
  color: #ffffff;
  :hover {
    color: #ffe500;
    cursor: pointer;
  }
",
                        "toString": [Function],
                      }
                    }
                    href="https://thegulocal.com/index/contributors"
                  >
                    All writers
                  </a>
                </li>
                <li
                  css={
                    Object {
                      "map": undefined,
                      "name": "66iazq",
                      "next": undefined,
                      "styles": "
  list-style: none;
",
                      "toString": [Function],
                    }
                  }
                >
                  <a
                    css={
                      Object {
                        "map": undefined,
                        "name": "1x52xav",
                        "next": undefined,
                        "styles": "
  display: inline-block;
  padding: 6px 0;
  color: #ffffff;
  :hover {
    color: #ffe500;
    cursor: pointer;
  }
",
                        "toString": [Function],
                      }
                    }
                    href="https://uploads.guim.co.uk/2021/07/27/STL_&_GMG_Modern_Slavery_Act_Statement_2021.pdf"
                  >
                    Modern Slavery Act
                  </a>
                </li>
                <li
                  css={
                    Object {
                      "map": undefined,
                      "name": "66iazq",
                      "next": undefined,
                      "styles": "
  list-style: none;
",
                      "toString": [Function],
                    }
                  }
                >
                  <a
                    css={
                      Object {
                        "map": undefined,
                        "name": "1x52xav",
                        "next": undefined,
                        "styles": "
  display: inline-block;
  padding: 6px 0;
  color: #ffffff;
  :hover {
    color: #ffe500;
    cursor: pointer;
  }
",
                        "toString": [Function],
                      }
                    }
                    href="https://theguardian.newspapers.com/"
                  >
                    Digital newspaper archive
                  </a>
                </li>
                <li
                  css={
                    Object {
                      "map": undefined,
                      "name": "66iazq",
                      "next": undefined,
                      "styles": "
  list-style: none;
",
                      "toString": [Function],
                    }
                  }
                >
                  <a
                    css={
                      Object {
                        "map": undefined,
                        "name": "1x52xav",
                        "next": undefined,
                        "styles": "
  display: inline-block;
  padding: 6px 0;
  color: #ffffff;
  :hover {
    color: #ffe500;
    cursor: pointer;
  }
",
                        "toString": [Function],
                      }
                    }
                    href="https://www.facebook.com/theguardian"
                  >
                    Facebook
                  </a>
                </li>
                <li
                  css={
                    Object {
                      "map": undefined,
                      "name": "66iazq",
                      "next": undefined,
                      "styles": "
  list-style: none;
",
                      "toString": [Function],
                    }
                  }
                >
                  <a
                    css={
                      Object {
                        "map": undefined,
                        "name": "1x52xav",
                        "next": undefined,
                        "styles": "
  display: inline-block;
  padding: 6px 0;
  color: #ffffff;
  :hover {
    color: #ffe500;
    cursor: pointer;
  }
",
                        "toString": [Function],
                      }
                    }
                    href="https://twitter.com/guardian"
                  >
                    Twitter
                  </a>
                </li>
              </ul>
              <ul
                css={
                  Object {
                    "map": undefined,
                    "name": "1hkg2v9",
                    "next": undefined,
                    "styles": "
  line-height: 19.2px;
  width: calc(50% - 1.25rem - 1px);
  list-style: none;
  position: relative;
  padding: 0 10px;
  margin: 0;

  &:nth-of-type(even) {
    border-left: 1px solid rgba(255, 255, 255, 0.3);
  }

  @media (min-width: 740px) {
    &:not(:first-of-type) {
      border-left: 1px solid rgba(255, 255, 255, 0.3);
    }

    width: 161px;
    flex: 1 0 0;
  }

  @media (min-width: 980px) {
    border-left: 1px solid rgba(255, 255, 255, 0.3);
  }
",
                    "toString": [Function],
                  }
                }
              >
                <li
                  css={
                    Object {
                      "map": undefined,
                      "name": "66iazq",
                      "next": undefined,
                      "styles": "
  list-style: none;
",
                      "toString": [Function],
                    }
                  }
                >
                  <a
                    css={
                      Object {
                        "map": undefined,
                        "name": "1x52xav",
                        "next": undefined,
                        "styles": "
  display: inline-block;
  padding: 6px 0;
  color: #ffffff;
  :hover {
    color: #ffe500;
    cursor: pointer;
  }
",
                        "toString": [Function],
                      }
                    }
                    href="https://advertising.thegulocal.com"
                  >
                    Advertise with us
                  </a>
                </li>
                <li
                  css={
                    Object {
                      "map": undefined,
                      "name": "66iazq",
                      "next": undefined,
                      "styles": "
  list-style: none;
",
                      "toString": [Function],
                    }
                  }
                >
                  <a
                    css={
                      Object {
                        "map": undefined,
                        "name": "1x52xav",
                        "next": undefined,
                        "styles": "
  display: inline-block;
  padding: 6px 0;
  color: #ffffff;
  :hover {
    color: #ffe500;
    cursor: pointer;
  }
",
                        "toString": [Function],
                      }
                    }
                    href="https://thegulocal.com/guardian-labs"
                  >
                    Guardian Labs
                  </a>
                </li>
                <li
                  css={
                    Object {
                      "map": undefined,
                      "name": "66iazq",
                      "next": undefined,
                      "styles": "
  list-style: none;
",
                      "toString": [Function],
                    }
                  }
                >
                  <a
                    css={
                      Object {
                        "map": undefined,
                        "name": "1x52xav",
                        "next": undefined,
                        "styles": "
  display: inline-block;
  padding: 6px 0;
  color: #ffffff;
  :hover {
    color: #ffe500;
    cursor: pointer;
  }
",
                        "toString": [Function],
                      }
                    }
                    href="https://jobs.thegulocal.com/?INTCMP=NGW_FOOTER_UK_GU_JOBS"
                  >
                    Search jobs
                  </a>
                </li>
                <li
                  css={
                    Object {
                      "map": undefined,
                      "name": "66iazq",
                      "next": undefined,
                      "styles": "
  list-style: none;
",
                      "toString": [Function],
                    }
                  }
                >
                  <a
                    css={
                      Object {
                        "map": undefined,
                        "name": "1x52xav",
                        "next": undefined,
                        "styles": "
  display: inline-block;
  padding: 6px 0;
  color: #ffffff;
  :hover {
    color: #ffe500;
    cursor: pointer;
  }
",
                        "toString": [Function],
                      }
                    }
                    href="https://patrons.thegulocal.com/?INTCMP=footer_patrons"
                  >
                    Patrons
                  </a>
                </li>
              </ul>
              <div
                css={
                  Object {
                    "map": undefined,
                    "name": "pt8jss",
                    "next": undefined,
                    "styles": "
  width: 50%;
  border-left: 1px solid rgba(255, 255, 255, 0.3);
  padding-left: 10px;
  @media (min-width: 660px) {
    width: 300px;
  }
  border-top: 1px solid rgba(255, 255, 255, 0.3);
  @media (min-width: 740px) {
    border-top: 0;
  }
",
                    "toString": [Function],
                  }
                }
              >
                <div
                  css={
                    Object {
                      "map": undefined,
                      "name": "1k0upo6",
                      "next": undefined,
                      "styles": "
  color: #ffe500;
  font-family: GH Guardian Headline, \\"GuardianTextEgyptian\\", Georgia, serif;
  font-size: 24px;
  line-height: 24px;
  font-weight: bold;
  margin-top: 3px;
  margin-bottom: 12px;
  @media (min-width: 660px) {
    font-size: 32px;
    line-height: 32px;
  }
",
                      "toString": [Function],
                    }
                  }
                >
                  Support The Guardian
                </div>
                <div
                  css={
                    Object {
                      "map": undefined,
                      "name": "l2bq3g",
                      "next": undefined,
                      "styles": "
  display: inline-block;
  margin-right: 10px;
  margin-bottom: 6px;
",
                      "toString": [Function],
                    }
                  }
                >
                  <a
                    href="https://support.thegulocal.com/contribute?INTCMP=mma_footer_support_contribute&acquisitionData=%7B%22source%22%3A%22GUARDIAN_WEB%22%2C%22componentType%22%3A%22ACQUISITIONS_MANAGE_MY_ACCOUNT%22%2C%22componentId%22%3A%22mma_footer_support_contribute%22%7D&displayExistingPaymentOptions=true"
                    onClick={[Function]}
                  >
                    <button
                      css={
                        Object {
                          "map": undefined,
                          "name": "1dykxcg",
                          "next": undefined,
                          "styles": "font-size:16px;font-family:\\"Guardian Text Sans Web\\", \\"Helvetica Neue\\", Helvetica, Arial, \\"Lucida Grande\\", sans-serif;border-radius:1000px;align-items:center;position:relative;:active{outline:none;}min-height:36px;height:36px;font-weight:bold;display:inline-flex;background:#FFE500;color:#121212;border:none;padding:1px 40px 0 18px;svg{fill:currentColor;height:34px;position:absolute;right:0;top:50%;transform:translate(0, -50%);transition:transform .3s, background .3s;width:36px;}:hover{background:hsl(53.89999999999998, 100%, 45%);svg{transform:translate(3px, -50%);}}cursor:pointer;max-width:calc(100vw - 40px);",
                          "toString": [Function],
                        }
                      }
                      onMouseUp={[Function]}
                    >
                      Contribute
                      <svg
                        viewBox="0 0 30 30"
                      >
                        <path
                          d="M22.8 14.6L15.2 7l-.7.7 5.5 6.6H6v1.5h14l-5.5 6.6.7.7 7.6-7.6v-.9"
                        />
                      </svg>
                    </button>
                  </a>
                </div>
                <a
                  href="https://support.thegulocal.com/subscribe?INTCMP=mma_footer_support_subscribe&acquisitionData=%7B%22source%22%3A%22GUARDIAN_WEB%22%2C%22componentType%22%3A%22ACQUISITIONS_MANAGE_MY_ACCOUNT%22%2C%22componentId%22%3A%22mma_footer_support_subscribe%22%7D&displayExistingPaymentOptions=true"
                  onClick={[Function]}
                >
                  <button
                    css={
                      Object {
                        "map": undefined,
                        "name": "1dykxcg",
                        "next": undefined,
                        "styles": "font-size:16px;font-family:\\"Guardian Text Sans Web\\", \\"Helvetica Neue\\", Helvetica, Arial, \\"Lucida Grande\\", sans-serif;border-radius:1000px;align-items:center;position:relative;:active{outline:none;}min-height:36px;height:36px;font-weight:bold;display:inline-flex;background:#FFE500;color:#121212;border:none;padding:1px 40px 0 18px;svg{fill:currentColor;height:34px;position:absolute;right:0;top:50%;transform:translate(0, -50%);transition:transform .3s, background .3s;width:36px;}:hover{background:hsl(53.89999999999998, 100%, 45%);svg{transform:translate(3px, -50%);}}cursor:pointer;max-width:calc(100vw - 40px);",
                        "toString": [Function],
                      }
                    }
                    onMouseUp={[Function]}
                  >
                    Subscribe
                    <svg
                      viewBox="0 0 30 30"
                    >
                      <path
                        d="M22.8 14.6L15.2 7l-.7.7 5.5 6.6H6v1.5h14l-5.5 6.6.7.7 7.6-7.6v-.9"
                      />
                    </svg>
                  </button>
                </a>
              </div>
            </div>
          </div>
          <div
            css={
              Object {
                "map": undefined,
                "name": "1wu2ni1",
                "next": undefined,
                "styles": "
  padding-bottom: 24px;
  padding-left: 20px;
  padding-right: 20px;
  position: relative;
",
                "toString": [Function],
              }
            }
          >
            <a
              css={
                Object {
                  "map": undefined,
                  "name": "rhcyrc",
                  "next": undefined,
                  "styles": "
  font-size: 16px;
  color: #ffffff;
  font-weight: bold;
  padding: 0 5px;
  background-color: #052962;
  :hover {
    color: #ffe500;
  }
  position: absolute;
  right: 20px;
  transform: translateY(-50%);
",
                  "toString": [Function],
                }
              }
              href="#top"
            >
              <span
                css={
                  Object {
                    "map": undefined,
                    "name": "m3p6yb",
                    "next": undefined,
                    "styles": "
  display: inline-block;
  padding-right: 5px;
  padding-top: 9px;
",
                    "toString": [Function],
                  }
                }
              >
                Back to top
              </span>
              <span
                css={
                  Object {
                    "map": undefined,
                    "name": "n17lfs",
                    "next": undefined,
                    "styles": "
  position: relative;
  float: right;
  background-color: currentColor;
  border-radius: 100%;
  height: 42px;
  width: 42px;
",
                    "toString": [Function],
                  }
                }
              >
                <span
                  css={
                    Object {
                      "map": undefined,
                      "name": "1e08lku",
                      "next": undefined,
                      "styles": "
  position: absolute;
  fill: #052962;
  top: 9px;
  left: 9px;
",
                      "toString": [Function],
                    }
                  }
                >
                  <svg
                    height="18"
                    viewBox="0 0 24 18"
                    width="24"
                  >
                    <path
                      d="M.4 15.3l10.5-8.4L12 6l1.1.9 10.5 8.4-.5.7L12 9.7.9 16l-.5-.7z"
                    />
                  </svg>
                </span>
              </span>
            </a>
            <div
              css={
                Object {
                  "map": undefined,
                  "name": "n9fado",
                  "next": undefined,
                  "styles": "
  @media (min-width: 740px) {
    padding-top: 6px;
  }
  padding-top: 26px;
  font-size: 12px;
",
                  "toString": [Function],
                }
              }
            >
              © 
              2022
               Guardian News and Media Limited or its affiliated companies. All rights reserved.
            </div>
          </div>
        </div>
      </div>
    </div>
  </footer>
</div>
`;
